### PR TITLE
RA-576

### PR DESCRIPTION
**Pull request by jdegraft**
Added proficient locales to user defaults

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java [hunk 1]
```diff
@@ -81,12 +140,149 @@ public void setDefaultLocale(String defaultLocale) {
...
+            setProficientLocalesList(proficientLocalesList);
+        }
+
+        public Set<LocaleOption> getProficientLocalesSet() {
+            return proficientLocalesSet;
+        }
+
+        public void setProficientLocalesSet(Set<LocaleOption> proficientLocalesSet) {
+            this.proficientLocalesSet = proficientLocalesSet;
+            if (displayLocales != null)
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java:167]: Our code convention is that the open brace should be on the same line as the if clause, though i would assume the formatter would fix it when building the project.

File: omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java [hunk 2]
```diff
@@ -54,25 +70,68 @@ public String post(PageModel model, @BindParams UserDefaults userDefaults,
...
 
+        public void setLanguage(String language) {
+            this.language = language;
+        }
+    }
+    
+    public class UserDefaults {
+
+        private String defaultLocale;
+        private Set<LocaleOption> proficientLocalesSet;
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java:129]: The options to select a default locale and proficient locales should be the same list  i.e the ones returned by AdministrationService.getPresentationLocales()

File: omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java [hunk 3]
```diff
@@ -54,25 +70,68 @@ public String post(PageModel model, @BindParams UserDefaults userDefaults,
...
 
-        private String defaultLocale;
-        private String proficientLocales;
+        private Locale locale;
+        private boolean selected;
+        private String name;
+        private String language;
 
-        public UserDefaults() {
+        public LocaleOption() {
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java:83]: Do you really need this? Just label and code should be all you need where the code is the language or language and country

File: omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java [hunk 4]
```diff
@@ -81,12 +140,149 @@ public void setDefaultLocale(String defaultLocale) {
...
+        private String getProficientLocalesFromSet() {
+        	if (proficientLocalesSet == null) 
+        		return "";
+        	StringBuilder retProficientLocales = new StringBuilder();
+            for (LocaleOption locOption : proficientLocalesSet)
+            {
+            	if (locOption.selected)
+            	{
+                	if (retProficientLocales.length() > 0)
+                		retProficientLocales = retProficientLocales.append(",");
```
Comment (wluyima) [inline omod/src/main/java/org/openmrs/module/adminui/page/controller/myaccount/ChangeDefaultsPageController.java:246]: The cleaner way to handle this is to add a StringToLocaleConverter to uiframework module and there examples of these in the module, field binding by type using the @BindParam annotation would create for you this list and set on the UserDefault Object magically with minimal code written

**Comment by wluyima** [comment]
This pull request appears invalid, it should from your fork to openmrs/master

**Comment by jdegraft** [comment]
I will close the pull request and create a new one when I get the changes following your review done.
This pull request was done in GitHub windows client and hence the wrong setup.

**Comment by wluyima** [comment]
Okay

**Pull request closed without merging**